platform/MCU/nRF52840/inc/dct.h

```diff
@@ -151,50 +151,57 @@ typedef struct __attribute__((packed)) application_dct {
+#define DCT_OTA_UPDATE_FLAG_SIZE  (sizeof((application_dct_t*)0)->ota_update_flag)
+#define DCT_FEATURE_FLAGS_SIZE  (sizeof((application_dct_t*)0)->feature_flags)
+#define DCT_COUNTRY_CODE_SIZE  (sizeof((application_dct_t*)0)->country_code)
+#define DCT_CLAIM_CODE_SIZE  (sizeof((application_dct_t*)0)->claim_code)
+#define DCT_SSID_PREFIX_SIZE  (sizeof((application_dct_t*)0)->ssid_prefix)
+#define DCT_DNS_RESOLVE_SIZE  (sizeof((application_dct_t*)0)->dns_resolve)
+#define DCT_DEVICE_CODE_SIZE  (sizeof((application_dct_t*)0)->device_code)
+#define DCT_DEVICE_CLAIMED_SIZE  (sizeof((application_dct_t*)0)->claimed)
+#define DCT_FLASH_MODULES_SIZE  (sizeof((application_dct_t*)0)->flash_modules)
+#define DCT_PRODUCT_STORE_SIZE  (sizeof((application_dct_t*)0)->product_store)
+#define DCT_ANTENNA_SELECTION_SIZE  (sizeof((application_dct_t*)0)->antenna_selection)
+#define DCT_CLOUD_TRANSPORT_SIZE  (sizeof((application_dct_t*)0)->cloud_transport)
+#define DCT_ALT_DEVICE_PUBLIC_KEY_SIZE  (sizeof((application_dct_t*)0)->alt_device_public_key)
+#define DCT_ALT_DEVICE_PRIVATE_KEY_SIZE  (sizeof((application_dct_t*)0)->alt_device_private_key)
+#define DCT_ALT_SERVER_PUBLIC_KEY_SIZE  (sizeof((application_dct_t*)0)->alt_server_public_key)
+#define DCT_ALT_SERVER_ADDRESS_SIZE  (sizeof((application_dct_t*)0)->alt_server_address)
+#define DCT_DEVICE_ID_SIZE  (sizeof((application_dct_t*)0)->device_id)
+#define DCT_RADIO_FLAGS_SIZE  (sizeof((application_dct_t*)0)->radio_flags)
+#define DCT_MODE_BUTTON_MIRROR_SIZE (sizeof((application_dct_t*)0)->mode_button_mirror)
+#define DCT_LED_MIRROR_SIZE (sizeof((application_dct_t*)0)->led_mirror)
+#define DCT_LED_THEME_SIZE (sizeof((application_dct_t*)0)->led_theme)
+#define DCT_EAP_CONFIG_SIZE (sizeof((application_dct_t*)0)->eap_config)
+#define DCT_DEVICE_SECRET_SIZE (sizeof((application_dct_t*)0)->device_secret)
+// #define DCT_SETUP_DONE_SIZE (sizeof((application_dct_t*)0)->setup_done)
+#define DCT_NCP_ID_SIZE (sizeof((application_dct_t*)0)->ncp_id)
+#define DCT_POWER_CONFIG_SIZE (sizeof((application_dct_t*)0)->power_config)
+#define DCT_RADIO_ANTENNA_SIZE (sizeof((application_dct_t*)0)->radio_antenna)
 
 #define STATIC_ASSERT_DCT_OFFSET(field, expected) PARTICLE_STATIC_ASSERT( dct_##field, offsetof(application_dct_t, field)==expected)
 #define STATIC_ASSERT_FLAGS_OFFSET(field, expected) PARTICLE_STATIC_ASSERT( dct_sysflag_##field, offsetof(platform_system_flags_t, field)==expected)
 
 #define DCT_OTA_UPDATE_FLAG_SET (0xA5)
 #define DCT_OTA_UPDATE_FLAG_CLEAR (0XFF)
 
+typedef enum platform_dct_deprecated {
+    DCT_SETUP_DONE_OFFSET __attribute__((deprecated("Setup Done flag has been deprecated since Device OS 4.0 and has no effect on device behavior")))
+            = (offsetof(application_dct_t, setup_done)),
+    DCT_SETUP_DONE_SIZE __attribute__((deprecated("Setup Done flag has been deprecated since Device OS 4.0 and has no effect on device behavior")))
```

Comment:
Since we are planning on removing SETUP DONE in 5.x completely (which will cause a build error), can we add that notification to these deprecation messages?  Something like "Will be removed in 5.x and cause a build error!" ?